src/update/balrog.cpp

```diff
@@ -188,69 +176,18 @@ bool Balrog::checkSignature(Task* task, const QByteArray& x5uData,
 bool Balrog::validateSignature(const QByteArray& x5uData,
                                const QByteArray& updateData,
                                const QByteArray& signatureBlob) {
-#if defined(MZ_WINDOWS)
-  typedef void BalrogSetLogger(GoUintptr func);
-  typedef GoUint8 BalrogValidate(GoString x5uData, GoString updateData,
-                                 GoString signature, GoString rootHash,
-                                 GoString leafCertSubject);
-
-  static HMODULE balrogDll = nullptr;
-  static BalrogSetLogger* balrogSetLogger = nullptr;
-  static BalrogValidate* balrogValidate = nullptr;
-
-  if (!balrogDll) {
-    balrogDll = LoadLibrary(TEXT("balrog.dll"));
-    if (!balrogDll) {
-      WindowsUtils::windowsLog("Failed to load balrog.dll");
-      return false;
-    }
-  }
-
-  if (!balrogSetLogger) {
-    balrogSetLogger =
-        (BalrogSetLogger*)GetProcAddress(balrogDll, "balrogSetLogger");
-    if (!balrogSetLogger) {
-      WindowsUtils::windowsLog("Failed to get balrogSetLogger function");
-      return false;
-    }
-  }
-
-  if (!balrogValidate) {
-    balrogValidate =
-        (BalrogValidate*)GetProcAddress(balrogDll, "balrogValidate");
-    if (!balrogValidate) {
-      WindowsUtils::windowsLog("Failed to get balrogValidate function");
-      return false;
-    }
-  }
-#endif
-
-  balrogSetLogger((GoUintptr)balrogLogger);
-
-  QByteArray x5uDataCopy = x5uData;
-  GoString x5uDataGo{x5uDataCopy.constData(), x5uDataCopy.length()};
-
-  QByteArray signatureCopy = signatureBlob;
-  GoString signatureGo{signatureCopy.constData(), signatureCopy.length()};
-
-  QByteArray updateDataCopy = updateData;
-  GoString updateDataGo{updateDataCopy.constData(), updateDataCopy.length()};
-
-  QByteArray rootHashCopy = Constants::AUTOGRAPH_ROOT_CERT_FINGERPRINT;
-  rootHashCopy = rootHashCopy.toUpper();
-  GoString rootHashGo{rootHashCopy.constData(), rootHashCopy.length()};
-
-  QByteArray certSubjectCopy = BALROG_CERT_SUBJECT_CN;
-  GoString certSubjectGo{certSubjectCopy.constData(), certSubjectCopy.length()};
-
-  unsigned char verify = balrogValidate(x5uDataGo, updateDataGo, signatureGo,
-                                        rootHashGo, certSubjectGo);
+  // TODO: It would be nice to print the error using logger().
+  bool verify = verify_balrog(
```

Comment:
Does this change mean that balrog.dll can be removed in Windows?

Reply:
It does! I have yet to dig out my Windows machine and test that the update still works, so I was hesitant to remove it from the build just yet. I should go do that.

Reply:
Turns out it works just fine! I opted to remove the DLL.